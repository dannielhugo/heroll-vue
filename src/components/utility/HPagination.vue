<template>
  <el-pagination
    layout="prev, pager, next, jumper"
    :total="total"
    :disabled="loading"
    :page-size="pageSize"
    @current-change="onPageChanged"
    @next-click="onPageChanged"
    @prev-click="onPageChanged"
  />
</template>

<script setup lang="ts">
const emit = defineEmits(['page-changed']);

defineProps<{
  total: number;
  loading: boolean;
  pageSize: number;
  currentPage: number;
}>();

const onPageChanged = (p: number) => {
  emit('page-changed', p);
};
</script>

<style scoped></style>
