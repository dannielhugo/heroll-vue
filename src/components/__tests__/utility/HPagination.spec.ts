import HPagination from '@/components/utility/HPagination.vue';
import ElementPlus from 'element-plus';
import { mount } from '@vue/test-utils';

describe('HPagination', () => {
  test('should mount component', async () => {
    expect(HPagination).toBeTruthy();

    const wrapper = mount(HPagination, {
      props: {
        loading: false,
        total: 100,
        pageSize: 10,
        currentPage: 1,
      },
      global: {
        plugins: [ElementPlus],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should emit event on page change', async () => {
    const wrapper = mount(HPagination, {
      props: {
        loading: false,
        total: 100,
        pageSize: 10,
        currentPage: 1,
      },
      global: {
        plugins: [ElementPlus],
      },
    });

    wrapper.vm.onPageChanged(1);
    expect(wrapper.emitted('page-changed')).toBeTruthy();
  })
});
