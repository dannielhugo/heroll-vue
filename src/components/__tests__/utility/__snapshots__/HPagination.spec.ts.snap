// Vitest Snapshot v1

exports[`HPagination > should mount component 1`] = `
"<div role=\\"pagination\\" aria-label=\\"pagination\\" class=\\"el-pagination\\"><button type=\\"button\\" class=\\"btn-prev is-first\\" disabled=\\"\\" aria-disabled=\\"true\\"><i class=\\"el-icon\\"><svg viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\">
        <path fill=\\"currentColor\\" d=\\"M609.408 149.376 277.76 489.6a32 32 0 0 0 0 44.672l331.648 340.352a29.12 29.12 0 0 0 41.728 0 30.592 30.592 0 0 0 0-42.752L339.264 511.936l311.872-319.872a30.592 30.592 0 0 0 0-42.688 29.12 29.12 0 0 0-41.728 0z\\"></path>
      </svg></i></button>
  <ul class=\\"el-pager\\">
    <li class=\\"is-active number\\" aria-current=\\"true\\" tabindex=\\"0\\"> 1 </li>
    <!--v-if-->
    <li class=\\"number\\" aria-current=\\"false\\" tabindex=\\"0\\">2</li>
    <li class=\\"number\\" aria-current=\\"false\\" tabindex=\\"0\\">3</li>
    <li class=\\"number\\" aria-current=\\"false\\" tabindex=\\"0\\">4</li>
    <li class=\\"number\\" aria-current=\\"false\\" tabindex=\\"0\\">5</li>
    <li class=\\"number\\" aria-current=\\"false\\" tabindex=\\"0\\">6</li>
    <li class=\\"more btn-quicknext el-icon\\" tabindex=\\"0\\"><svg viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\">
        <path fill=\\"currentColor\\" d=\\"M176 416a112 112 0 1 1 0 224 112 112 0 0 1 0-224zm336 0a112 112 0 1 1 0 224 112 112 0 0 1 0-224zm336 0a112 112 0 1 1 0 224 112 112 0 0 1 0-224z\\"></path>
      </svg></li>
    <li class=\\"number\\" aria-current=\\"false\\" tabindex=\\"0\\">10</li>
  </ul><button type=\\"button\\" class=\\"btn-next\\" aria-disabled=\\"false\\"><i class=\\"el-icon\\"><svg viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\">
        <path fill=\\"currentColor\\" d=\\"M340.864 149.312a30.592 30.592 0 0 0 0 42.752L652.736 512 340.864 831.872a30.592 30.592 0 0 0 0 42.752 29.12 29.12 0 0 0 41.728 0L714.24 534.336a32 32 0 0 0 0-44.672L382.592 149.376a29.12 29.12 0 0 0-41.728 0z\\"></path>
      </svg></i></button><span class=\\"el-pagination__jump is-last\\" disabled=\\"false\\">Go to <div class=\\"el-input el-input--small el-pagination__editor is-in-pagination\\"><!-- input --><!-- prepend slot --><!--v-if--><div class=\\"el-input__wrapper\\"><!-- prefix slot --><!--v-if--><input class=\\"el-input__inner\\" min=\\"1\\" max=\\"10\\" type=\\"number\\" autocomplete=\\"off\\" tabindex=\\"0\\"><!-- suffix slot --><!--v-if--></div><!-- append slot --><!--v-if--></div> </span>
</div>"
`;
